src/lib/nvme_pt_ioctl.c

```diff
@@ -1566,7 +1566,7 @@ static struct opal_req_item opal_read_datastr_cmd[] = {
 static int opal_read_datastr(int fd, struct opal_device *dev, uint8_t *data, uint64_t offset, uint64_t size)
 {
 	int ret = 0;
-	uint64_t len = 0, index = 0;
+	uint64_t len = 0, index = 0, end_row;
```

Comment:
Declare and initialize end_row = size - 1;

Reply:
fixed